### PR TITLE
feat(ingest/lookml): CLL support for sql set in sql_table_name attribute of lookml view

### PR DESCRIPTION
**Pull request by sid-acryl**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new SQL view for analyzing rental income as a source of employee income.
  - Added new functionalities for improved handling of SQL patterns and derived views in Looker integration.

- **Enhancements**
  - Enhanced the `LookerViewContext` class for better SQL query classification.
  - Streamlined SQL query processing through modular functions for cleaner code.
  - Improved dataset metadata with additional aspects for better usability and data management.

- **Bug Fixes**
  - Addressed completeness issues in metadata processing related to Looker source integration.

- **Documentation**
  - Expanded documentation to include updated patterns for SQL table definitions within Looker views.

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/looker/lookml_concept_context.py [hunk 1]
```diff
@@ -413,3 +455,14 @@ def is_sql_based_derived_view_without_fields_case(self) -> bool:
             return True
 
         return False
+
+    def is_direct_sql_query_case(self) -> bool:
+        # pattern 7
+        # sqlglot doesn't have a function to validate whether text is valid SQL or not.
+        # Applying a simple logic to check if sql_table_name contains a sql.
+        # if sql_table_name contains sql then its value starts with "(" and checking if "select" is present in side the
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/looker/lookml_concept_context.py:463]: sql doesn't need to start with `(`

or is this something specific to lookml sql_table_name?

File: metadata-ingestion/src/datahub/ingestion/source/looker/lookml_concept_context.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 import re
 from typing import Any, Dict, List, Optional
 
+import sqlglot
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/looker/lookml_concept_context.py:6]: this import is unused right?

File: metadata-ingestion/src/datahub/ingestion/source/looker/lookml_concept_context.py (hunk 1 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/looker/lookml_concept_context.py:463]: specific to sql_table_name

File: metadata-ingestion/src/datahub/ingestion/source/looker/lookml_concept_context.py (hunk 2 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/looker/lookml_concept_context.py:6]: right, will remove it

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**